src/platform/gl/sdl_gl_platform.cpp

```diff
@@ -0,0 +1,84 @@
+#include "platform/platform.h"
+#include "common/error.h"
+
+#include "SDL.h"
+#include "GL/gl3w.h"
+
+struct gl_device
```

Comment:
In c++, placing structs, classes, and functions in an anonymous namespace (even a nested one) is the same as declaring a function static in C. I think it's a nice of way documenting which types and functions are meant to be translation-unit local 🙂